update-doi-request/src/test/java/no/unit/nva/doi/requests/handlers/ApiUpdateDoiRequestHandlerTest.java

```diff
@@ -0,0 +1,341 @@
+        assertThat(errorMessage, not(containsString(username)));
+        assertThat(errorMessage, not(containsString(publicationOwner)));
+    }
+
+    private void assertThatLogsContainReasonForForbiddenMessage(TestAppender appender, Publication publication) {
+        String expectedErrorMessage = String.format(WRONG_OWNER_ERROR, INVALID_USERNAME, publication.getOwner());
+        assertThat(appender.getMessages(), containsString(expectedErrorMessage));
+    }
+
+    @Test
+    public void handleRequestSucessfullyUpdatesStatusWhenPublicationIdIsValid() throws IOException, NotFoundException {
+        var publication = insertPublicationWithDoiRequest(Clock.fixed(mockNow,
+                ZoneId.systemDefault()));
+
+
+        var updateRequest = new ApiUpdateDoiRequest();
+        updateRequest.setPublicationId(publication.getIdentifier().toString());
+        updateRequest.setDoiRequestStatus(DoiRequestStatus.APPROVED);
+
+
+        var output = outpuStream();
+
+        GatewayResponse<ApiUpdateDoiResponse> response = sendRequest(updateRequest, validUsername(publication));
+
+        assertThat(response.getStatusCode(), is(equalTo(HttpStatus.SC_ACCEPTED)));
+
+        DoiRequestSummary actualDoiRequestSummary = readPublicationDirectlyFromDynamo(publication.getIdentifier());
+        DoiRequestSummary expectedDoiRequestSummary = expectedDoiRequestSummary(publication);
+
+        assertThat(actualDoiRequestSummary, is(equalTo(expectedDoiRequestSummary)));
+
+        var responseEntity = response.getBodyObject(ApiUpdateDoiResponse.class);
+        assertThat(responseEntity, is(equalTo(ApiUpdateDoiResponse.newBuilder()
+            .withTask(ApiTask.newBuilder()
+                .withIdentifier(publication.getIdentifier().toString())
+                .withHref("https://mocked-hostname.example.net/publication/" + publication.getIdentifier().toString())
+                .build())
+            .build())));
+    }
+
+    private String validUsername(Publication publication) {
+        return publication.getOwner();
+    }
+
+    private DoiRequestSummary readPublicationDirectlyFromDynamo(UUID id) throws NotFoundException {
+        return doiRequestsService.fetchDoiRequest(id).orElseThrow();
+    }
+
+    private DoiRequestSummary expectedDoiRequestSummary(Publication publication) {
+        var includedDoiRequest = new DoiRequest.Builder()
+                .withDate(mockNow)
+                .withStatus(DoiRequestStatus.APPROVED)
+                .build();
+        return new DoiRequestSummary(
+                publication.getIdentifier(),
+                publication.getOwner(),
+                includedDoiRequest,
+                publication.getEntityDescription());
+    }
+
+    private Publication insertPublicationWithDoiRequest(Clock clock)
+        throws com.fasterxml.jackson.core.JsonProcessingException {
+        Publication publication = PublicationGenerator.getPublicationWithDoiRequest(clock)
```

Comment:
Better to fix the method getPublicationWithDoiRequest so that we do not set the modified date without setting the created date.

Reply:
Im trying to limit the scope of what we change within the task that I'am solving, NP-1046. This generator is used by many other tests as well.

Reply:
true. resolved.